f90wrap/f90wrapgen.py

```diff
@@ -331,6 +364,8 @@ def write_init_lines(self, node):
         """
         for alloc in node.allocate:
             self.write("allocate(%s_ptr%%p)" % alloc)  # (self.prefix, alloc))
+        if self.is_class(node.type_name) and node.name.endswith("_initialise"):
```

Comment:
@jameskermode this PR is ready for a rough review to get your opinion. This is a significant change for types and classes, as the code

1. Wraps all objects that are used as a `class` anywhere into another wrapper type
2. Allows to access member variables of classes via this nested type structure
3. Adds an allocate in the initialiser for the inner object (this is still a hack), and removes the explicit call to the `final ::` procedure in the finaliser, as this is handled via the destructor

Let me know what you think, especially how to fix the initialiser in a clean way. I will then revise it once more.

Reply:
TODO: fix in https://github.com/itpplasma/f90wrap/blob/c099fcfb602f0eea1cc3aa450539484b500db3bf/f90wrap/transform.py#L821